test/gtest/ucp/ucp_datatype.h

```diff
@@ -88,9 +89,10 @@ class data_type_desc_t {
     };
 
     bool is_valid() const {
-        return (m_buf != NULL) && (m_count != 0) &&
-               (UCP_DT_IS_IOV(m_dt) ? (m_count <= m_iov_cnt_limit) :
-               (UCP_DT_IS_CONTIG(m_dt) || UCP_DT_IS_GENERIC(m_dt)));
+        return (m_is_empty && (m_count == 0)) ||
+               ((m_buf != NULL) && (m_count != 0) &&
+                (UCP_DT_IS_IOV(m_dt) ? (m_count <= m_iov_cnt_limit) :
```

Comment:
pls remove is_empty, check that count==0||m_buf!=NULL

Reply:
and check that m_dt is contig/iov/generic

Reply:
this check is already here. We do not have invalid datatype, so can't use it as an indication of non-initialized desc.
That's why I added new boolean member